File: src/Homework1/Homework1Tests/ProgramTests.cs
﻿using Homework1;
using Xunit;

namespace Homework1Tests
{
    public class ProgramTests
    {
        [Fact]
        public void Main_WrongInput_WillReturnNotZero()
        {
            var args = new[] { "12", "+", "qu" };
            var result = Program.Main(args);
            Assert.True(result != 0);
        }
        
        [Fact]
        public void Main_CorrectInput_WillReturnZero()
        {
            var args = new[] { "12", "+", "34" };
            var result = Program.Main(args);
            Assert.True(result == 0);
        }
    }
}

File: src/Homework4/Homework4Tests/ProgramTests.fs
module Tests.ProgramTests

open Homework4
open Xunit

[<Fact>]
let Main_WrongInputData_WillReturnNotZero () =
    let args = [|"12";"+";"wvw23"|]
    let check = Program.main args
    Assert.True(check <> 0)

[<Fact>]
let Main_CorrectInputData_WillReturnZero () =
    let args = [|"12";"+";"34"|]
    let check = Program.main args
    let result = check = 0
    Assert.True(result)     